{{cookiecutter.project_slug}}/requirements/requirements.txt

```diff
@@ -1,3 +1 @@
-{%- if cookiecutter.sample_code != 'no' -%}
```

Comment:
@regDaniel Indeed, the `*requirements.txt` are essentially replaced by `*environment.yml` files with the switch to conda and can thus be removed. (Doesn't necessarily have to be done in this PR, though.)

Reply:
Just updated a PR for pytrajplot today where I very much reduced the Makefile to only the essential tasks. Here my experience concerning requirement specification, which might have interesting implications for the blueprint as well, especially regarding the name of the requirements file:

- `conda create` only accepts a simple list of packages in a text file (exactly one package name and, if necessary, the version specification per line, no dash, no lines with colon) and fails with yaml files
- `conda env create` on the contrary accepts both yaml and simple lists but requires the input file to have either a .yaml, .yml or .txt extension. All others file extensions, namely requirements**.in**, are rejected!

My conclusion was to have a requirements.txt (not requirements.in) file with unpinned requirements and an environment.yml file for pinned requirements, both for use with `conda env create`. 

BTW For pytrajplot, the appropriate file is chosen depending on the `PINNED ` variable that can be set to 0 for unpinned and to 1 (the default) for pinned requirements in the call to make, as in `make install PINNED=0`. However as you are moving away from make, this is probably out of question for the blueprint.

Reply:
As far as I understand it, the purpose of `conda env create` is to exactly recreate existing environments, hence it's acceptence of `*environment.yml` and `*requirements.txt` files, which are the standard names for files with pinned dependencies. By contrast, `conda create` is the general-purpose way of creating an environment and thus accepts any files with dependencies, whether they can be expected be pinned or not. (If I remember correctly, I once read a discussion in a conda issue that concluded that `conda env create` is pretty much only retained for backward compatibility, though it's been a while don't have the link at hand...)

> My conclusion was to have a requirements.txt (not requirements.in) file with unpinned requirements

Since dependency specification in Python is very much a jungle, the `*requirements.in` naming is not a ubiquitous standard (but at least the way a fairly widely used tool (`pip-tools` I think?) does it this way). But what's for sure standard is using `requirements.txt` for pinned dependencies, so I wouldn't recommend using `.txt` for files with unpinned dependencies, as this goes actively against about the only standard there is.

Reply:
ok, so the conclusion is we still need requirements.in (for unpinned) and requirements.txt (for pinned)? 
Not sure if there is consensus here, but perhaps we can shortly discuss it otherwise sometime next week.
For the time being and for this PR, I didnt change the convention. I kept it as it is, but updated the pinned requirements (that were not working), and added all the tests in jenkins

Reply:
I do not think that we still need `requirements.txt` for the pinned installation or at least I would not see a reason why. If it's just about the env name for `conda create`, we'll find another way as the env name is in principle set by cookiecutter (or manually by the user) upon its first creation.

Reply:
follow-up: I see that the jenkins plan uses requirements.txt files for pinned installations. Of course it's also possible to drop the .yml files, i the generation of a requirements.txt is more straightforward or has other advantages, I guess this is up to @ruestefa to decide once he writes the env export script. But we should decide between the two and keep the installation process the same for jenkins and for the actual installation with the installation scripts.

Reply:
ok, let me see if I can summarize, @ruestefa  argues requirements.yml or requirements.txt for pinned dependencies and requirements.in for unpinned dependencies (since a more standard in pip).  @regDaniel argues for having requirements.yml for pinned dependencies. So I can move the requirements.txt to requirements.yml. 
The only reason @regDaniel the jenkins files are using `requirements.txt` is because   `requirements.yml` do not exist yet (as far as I know, isnt?) 

One more question, since we are using conda, and not pip, why not using requirements.yml then for both, pinned and unpinned '

Reply:
> since we are using conda, and not pip, why not using requirements.yml then for both, pinned and unpinned

The `*.yml` files are created with `conda env export`, which exports the whole environment (full dependency tree with pinned version numbers). I am not aware of a way to specify unpinned top-level dependencies in `*.yml`. Presumably, one could manually write a `*.yml` file following the format that conda exports and conda may accept that during environment creation, but this is really needlessly complicated as a plain text file is more human-friendly than yaml. There's also never any need to manually edit an `*environment.yml` file, so there's no benefit in using the same format for unpinned requirements and pinned environments.